crates/shared/src/price_estimation/zeroex.rs

```diff
@@ -12,21 +9,10 @@ use std::sync::Arc;
 
 pub struct ZeroExPriceEstimator {
     pub api: Arc<dyn ZeroExApi>,
-    pub bad_token_detector: Arc<dyn BadTokenDetecting>,
 }
 
 impl ZeroExPriceEstimator {
     async fn estimate(&self, query: &Query) -> Result<Estimate, PriceEstimationError> {
-        if query.buy_token == query.sell_token {
```

Comment:
Maybe let's still assert those assumptions (in case someone configures the estimator chain incorrectly)

Reply:
Done.